core/platform/email/mailgun_email_services.py

```diff
@@ -16,10 +16,39 @@
 
 """Provides mailgun api to send email."""
 
+import base64
+import urllib
+import urllib2
+
 from core.platform.email import gae_email_services
 import feconf
 
-import requests
+
+def post_to_mailgun(data):
+    """Send POST HTTP request to mailgun api. This method is adopted from
+    the requests library's post method.
+
+    Args:
+        - data: dict. The data to be sent in the request's body.
+
+    Returns:
+         Response from the server. The object is a file-like object.
+         https://docs.python.org/2/library/urllib2.html
+    """
+    if not feconf.MAILGUN_API_KEY:
+        raise Exception('Mailgun API key is not available.')
+
+    if not feconf.MAILGUN_DOMAIN_NAME:
+        raise Exception('Mailgun domain name is not set.')
+
+    encoded = base64.b64encode(b'api:%s' % feconf.MAILGUN_API_KEY).strip()
+    auth_str = 'Basic %s' % encoded
+    header = {'Authorization': auth_str}
+    server = (
+        'https://api.mailgun.net/v3/%s/messages' % feconf.MAILGUN_DOMAIN_NAME)
+    data = urllib.urlencode(data)
+    req = urllib2.Request(server, data, header)   #nosec
```

Comment:
I think this was due to the codacy issues, right? Just a couple of thoughts:

- If this is what `requests` is already doing, just remove #nosec (we are looking into how to get rid of that check, but don't worry, it won't block submission of this PR).
- On the other hand, if it is actually a security issue, then we should figure out what the issue is and how to fix it.

Reply:
Yes, this is for the codacy issue. This is one security issue since 
>urlib not only opens http:// or https:// URLs, but also ftp:// and file://. 
https://stackoverflow.com/a/53040523/7676306

The general way to fix this issue is to check the URL format in advance. Since curretnly, we are hardcoding the URL, so I think it should be fine. 

So, I will remove the comment.